spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/BoundStatementBuilder.scala

```diff
@@ -38,12 +68,14 @@ private[connector] class BoundStatementBuilder[T](
       val converter = converters(i)
       val columnName = columnNames(i)
       val columnValue = converter.convert(buffer(i))
-      boundStatement.set(columnName, columnValue, CodecRegistryUtil.codecFor(columnTypes(i), columnValue))
```

Comment:
Hmm I like this approach too. I wanted to make it clear that what portion of the build was different. I have no strong feelings either way I can switch it to this if you like? I'm not sure it makes much of a different performance wise since Hotspot should optimize this loop regardless of how we write it (i think)